android/src/main/java/com/mattermost/networkclient/NetworkClient.kt

```diff
@@ -183,8 +184,12 @@ class NetworkClient(private val baseUrl: HttpUrl? = null, private val options: R
         return okHttpClient.newCall(request)
     }
 
-    fun createWebSocket(listener: WebSocketEventListener) {
-        val request = Request.Builder().build()
+    fun createWebSocket(uri: URI) {
```

Comment:
@migbot I still find it very weird that the `baseUrl` and `webSocketUri` are configurable independently of each other. Shouldn't the `NetworkClient` have a single `baseUrl` which it uses for all connections? With WebSockets that would just mean internally replacing the scheme when connecting.

With the current implementation it's possible to instantiate a `NetworkClient` with `baseUrl = https://example.com/foo`, `webSocketUri = wss://something-entirely-different.example/bar`, and `trustSelfSignedServerCertificate = true`. And if you do that I have how I should expect it to behave.

Reply:
@jupenur I would expect ppl using this lib to be able to do exactly that, connect with the WS to a different server.

For example we may want to replace the WS used for development to connect the metro packager instead of using the one shipped with RN (we won't really be doing this but it is a possibility).

If we look at out current solutions on web and mobile they do have to set the url's separately as the WebSocket totally separate from fetch/XMLRequest, so the question is why would we want to prevent that.